### PR TITLE
Refactor CommandQueueMT

### PR DESCRIPTION
**Pull request by reduz**
* RingBuffer had no reason to be in this context, since everything is flushed as soon as possible. Flushing all elements is more efficient than flushing one by one (which required individual locking).
* A single buffer is used that can grow as much as the game needs.

This should make thread loading entirely reliable.

### CODE REVIEW

File: tests/test_command_queue.h [hunk 1]
```diff
@@ -276,50 +276,6 @@ TEST_CASE("[CommandQueue] Test Queue Basics") {
 			ProjectSettings::get_singleton()->property_get_revert(COMMAND_QUEUE_SETTING));
 }
```
Comment (reduz) [inline tests/test_command_queue.h:278]: For the record the test below is no longer needed because the queue cant go full any longer.

**Comment by HaSa1002** [comment]
Is #49201 unrelated to this? I tested with a build and it seems to dead lock in a different location now. So I am not sure.

**Comment by reduz** [comment]
@HaSa1002 it should not deadlock, the bug may be something else, but it will be worth checking eventually

**Review by pouleyKetchoupp — APPROVED** [review]
The code looks good, and physics on thread seems to still work fine.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**